src/components/input/TypeHandler/price.ts

```diff
@@ -0,0 +1,65 @@
+import { Base } from "./Base"
+import { Component } from "../Component"
+import { TypeHandler } from "./TypeHandler"
+import { StateEditor } from "../StateEditor"
+import { State } from "../State"
+
+class Price extends Base {
+	get native(): Component<string> {
+		const result = super.native
+		return {
+			...result,
+			type: "text",
+		}
+	}
+	constructor(component: Component<any>) {
+		super(component)
+	}
+	handleBlur(state: State, event: FocusEvent): State {
+		const stateEditor = StateEditor.copy(state)
```

Comment:
Why do the copy here every time it is implemented?

Reply:
What alternative do you suggest? Using .modify doesn't work. I'm passing down a readonly state from TypeHandler.